File: support/mkdb.pl
#!/usr/bin/env perl

###
## @(#) mkdb.pl    falcon@icb.net  2/1/98
## Updated by hoche@grok.com       5/24/19
##
## initialize the user database with the required "server" nickname
###

use strict;
use warnings;

sub promptUser {
   my ($promptString, $defaultValue) = @_;

   if ($defaultValue) {
      print $promptString, " [", $defaultValue, "]: ";
   } else {
      print $promptString, ": ";
   }

   $| = 1; # force flush
   $_ = <STDIN>;

   chomp;

   if ("$defaultValue") {
      return $_ ? $_ : $defaultValue;
   } else {
      return $_;
   }
}

my $REALNAME = &promptUser("The server admin's name", "Server Admin");
my $HOME = &promptUser("The server address", "icb\@yourdomain.com");
my $EMAIL = &promptUser("The server admin's email", "icbadmin\@yourdomain.com");

dbmopen(my %DB, "./icbdb", 0666);
$DB{ "server.realname"} = $REALNAME;
$DB{ "server.nick"} = "server";
$DB{ "server.home"} = $HOME;
$DB{ "server.email"} = $EMAIL;
$DB{ "server.text"} = "Here to server you!";
$DB{ "server.nummsg"} = "0";
$DB{ "server.www"} = "http://www.icb.net/";

dbmclose(%DB);

File: support/mkdb.py
#!/usr/bin/env python3

###
## mkdb.py    falcon@icb.net    2/1/98
##
## initialize the user database with the required "server" nickname
###

import sys
import os

sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
from icbdb import IcbDb

def prompt_user(prompt, default=None):
    if default:
        s = input(f"{prompt} [{default}]: ")
    else:
        s = input(f"{prompt}: ")
    if not s and default:
        return default
    return s

def main():
    dbpath = sys.argv[1] if len(sys.argv) > 1 else "./icbdb"

    realname = prompt_user("The server admin's name", "Server Admin")
    home     = prompt_user("The server address", "icb@yourdomain.com")
    email    = prompt_user("The server admin's email", "icbadmin@yourdomain.com")

    with IcbDb(dbpath) as db:
        db["server.realname"] = realname
        db["server.nick"]     = "server"
        db["server.home"]     = home
        db["server.email"]    = email
        db["server.text"]     = "Here to server you!"
        db["server.nummsg"]   = "0"
        db["server.www"]      = "http://www.icb.net/"

if __name__ == "__main__":
    main()
